### ISSUE TITLE
iPad rotation bug

### ISSUE THREAD

**Issue by phena109** [sapegin/jquery.mosaicflow#17]
Hi 

I had been fighting with this plugin bug for the past few days on ipad.

I have a list which expected to split to 2 columns in landscape and back to 1 column in portrait. But the result is that it will on every alternative change from portrait to landscape show only one column instead of 2.

I still can't find out why every alternative change from portrait to landscape. But I can see that you only remove the hidden column on the next visit to the refill() function and on the instance that show 1 column in landscape, desktop browser tend to fire refill() twice which will cover the incapability to render properly of the list in the first call of refill(). This however is not same situation for iPad. refill() only fire once and by the time refill() is able remove the hidden column it is already too late because all the items already got into the first column.

Long story short, eventually I came up with this solution which will solve my problem and possibly other people's:

At the first line of the ensureColumns() function, I will use the following line instead of the orginal one.
var createdCnt = this.columns.filter(':visible').length;

This is to filter out those are not viewable (anyway) and count those visible, before the unused column really get deleted.

Hop this can help
D.

**Comment by Barrytron** [comment]
Thanks you legend!
